test/aimsc_test.js

```diff
@@ -0,0 +1,188 @@
+const fs = require('fs');
+const assert = require('assert');
+const rewire = require('rewire');
+const sinon = require('sinon');
+const tk = require('timekeeper');
+const AimsC = require('../al_servicec').AimsC;
+const AzcollectC = require('../al_servicec').AzcollectC;
+const m_alMock = require('./al_mock');
+const debug = require('debug') ('azcollectc_test');
+var servicecRewire = rewire('../al_servicec');
+var m_servicec = require('../al_servicec');
+var RestServiceClient = require('../al_util').RestServiceClient;
+
+const INITIAL_TS = 1529572769;
+const BEFORE_EXPIRED = (INITIAL_TS + 600) * 1000;
+const AFTER_EXPIRED = (INITIAL_TS + m_alMock.AIMS_TOKEN_TTL - 600) * 1000;
+
+
+describe('Unit Tests', function() {
+
+    describe('AimsC', function() {
+        var fakeRest;
+
+        beforeEach(function() {
+            tk.freeze(new Date(INITIAL_TS * 1000));
+
+            fakeRest = sinon.stub(RestServiceClient.prototype, 'post').callsFake(
+                function fakeFn(path, options) {
+                    if (path == '/aims/v1/authenticate' &&
+                        options.auth.user == m_alMock.AIMS_AUTH.auth.user &&
+                        options.auth.password == m_alMock.AIMS_AUTH.auth.password) {
+                        return new Promise(function(resolve, reject) {
+                            resolve({
+                                authentication : {
+                                    token : 'token',
+                                    account : {
+                                        id : m_alMock.CID
+                                    },
+                                    token_expiration : Math.ceil(Date.now()/1000 + m_alMock.AIMS_TOKEN_TTL)
+                                }
+                            });
+                        });
+                    } else {
+                        return new Promise(function(resolve, reject) {
+                            reject('error');
+                        });
+                    }
+                }
+            );
+            clean_file_cache(m_alMock.CACHE_FILENAME);
+        });
+
+        afterEach(function() {
+            fakeRest.restore();
+            tk.reset();
+        });
+
+        after(function() {
+            clean_file_cache(m_alMock.CACHE_FILENAME);
+        });
+
+        it('authenticate with cold cache', function(done) {
```

Comment:
I am also going to add test about handling "bad credentials" case (when creds are rejected)